### PR TITLE
[kokkos] Rewrite Hist::finalize() with Teamscan

### PR DESCRIPTION
**Pull request by PointKernel**
I tried to fuse multiple kernels into one since `finalize()` is now a device function but got a sequence of different errors. Will do it next week.

### CODE REVIEW

File: src/kokkos/test/kokkos/OneHistoContainer_t.cc [hunk 1]
```diff
@@ -46,7 +46,7 @@ void go() {
 
     using TeamHist = cms::kokkos::HistoContainer<T, NBINS, N, S, uint16_t>;
 
-    Kokkos::View<TeamHist, KokkosExecSpace> histo_d("histo_d");
+    Kokkos::View<TeamHist[1], KokkosExecSpace> histo_d("histo_d");
```
Comment (makortel) [inline src/kokkos/test/kokkos/OneHistoContainer_t.cc:49]: Why was this change needed?

**Review by makortel — COMMENTED** [review]
Thanks, looks good. I have only one question.

**Comment by makortel** [comment]
`VertexFinder_t` unit test fails with
```
Running test /global/cscratch1/sd/matti/cce/prtest/pixeltrack-standalone/test/kokkos/VertexFinder_t.device
v,t size 26 301
M eps, pset 2 0.1 0
nt,nv 301 0,0
:0: : block: [0,0,0], thread: [0,32,0] Assertion `Cuda::cuda_intra_block_scan requires power-of-two blockDim` failed.
:0: : block: [0,0,0], thread: [0,33,0] Assertion `Cuda::cuda_intra_block_scan requires power-of-two blockDim` failed.
:0: : block: [0,0,0], thread: [0,34,0] Assertion `Cuda::cuda_intra_block_scan requires power-of-two blockDim` failed.
...
terminate called after throwing an instance of 'std::runtime_error'
  what():  cudaStreamSynchronize(m_stream) error( cudaErrorAssert): device-side assert triggered .../pixeltrack-standalone/external/kokkos/source/core/src/Cuda/Kokkos_Cuda_Instance.cpp:314
Traceback functionality not available
```

File: src/kokkos/test/kokkos/OneHistoContainer_t.cc (hunk 1 above)
Comment (PointKernel) [reply src/kokkos/test/kokkos/OneHistoContainer_t.cc:49]: In `Hist::finalize()`, each hist view has to take an argument, e.g. `update += histo(leagueRank).off[i];`. This expression cannot be applied to "scalar" view. For example,
```c++
Kokkos::View<TeamHist, KokkosExecSpace> histo_d("histo_d");
Kokkos:: parallel_scan( ...
  update += histo(0).off[i]; // this will not work, result in compile time error
  update += histo().off[i]; // this will work, "scalar" view takes no argument
);
```
Since the original `forEachInWindow` function works over a scalar view only and the view in `finalize()` has to take an argument, I just make a minimal change to make both of them run. Let me know if a thorough solution to this is required.

**Comment by PointKernel** [comment]
> `VertexFinder_t` unit test fails with
> 
> ```
> Running test .../pixeltrack-standalone/test/kokkos/VertexFinder_t.device
> v,t size 26 301
> M eps, pset 2 0.1 0
> nt,nv 301 0,0
> :0: : block: [0,0,0], thread: [0,32,0] Assertion `Cuda::cuda_intra_block_scan requires power-of-two blockDim` failed.
> :0: : block: [0,0,0], thread: [0,33,0] Assertion `Cuda::cuda_intra_block_scan requires power-of-two blockDim` failed.
> :0: : block: [0,0,0], thread: [0,34,0] Assertion `Cuda::cuda_intra_block_scan requires power-of-two blockDim` failed.
> ...
> terminate called after throwing an instance of 'std::runtime_error'
>   what():  cudaStreamSynchronize(m_stream) error( cudaErrorAssert): device-side assert triggered .../pixeltrack-standalone/external/kokkos/source/core/src/Cuda/Kokkos_Cuda_Instance.cpp:314
> Traceback functionality not available
> ```

File: src/kokkos/test/kokkos/OneHistoContainer_t.cc (hunk 1 above)
Comment (makortel) [reply src/kokkos/test/kokkos/OneHistoContainer_t.cc:49]: Thanks. Just to check that I understood, the need to be able to call `HistoContainer::finalize()` for an array of `HistoContainer` necessitates the indexing via `histo(leagueRank)`, and therefore all calls need to look like arrays (i.e. mimic a scalar with 1-element array). 

Then I'm wondering why
https://github.com/cms-patatrack/pixeltrack-standalone/blob/95d56aad8feb461165fe7563bedc564ac3c4f816/src/kokkos/KokkosCore/HistoContainer.h#L349
takes a scalar and not `Kokkos::View<Histo*, ExecSpace>`?

Comment (PointKernel) [reply src/kokkos/test/kokkos/OneHistoContainer_t.cc:49]: Oh no, that's a mistake. Thanks for pointing this out.
